### PR TITLE
Load Balancer new features

### PR DESCRIPTION
**Pull request by ddymko**
## Description
New LB features allow for attaching the LB to a private network and setting firewalls on the LB.

### Checklist:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you linted your code locally prior to submission?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: vultr/data_source_vultr_load_balancer.go [hunk 1]
```diff
@@ -170,5 +180,20 @@ func dataSourceVultrLoadBalancerRead(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("error setting `health_check`: %#v", err)
 	}
 
+	var fwrRules []map[string]interface{}
+	for _, rules := range lbList[0].FirewallRules {
+		rule := map[string]interface{}{
+			"id":      rules.ID,
```
Comment (mamclaughlin) [inline vultr/data_source_vultr_load_balancer.go]: should be RuleID

File: vultr/resource_vultr_load_balancer.go [hunk 2]
```diff
@@ -224,6 +255,14 @@ func resourceVultrLoadBalancerCreate(ctx context.Context, d *schema.ResourceData
 		stickySessions = nil
 	}
 
+	var fwrMap []govultr.LBFirewallRule
+	if firewallRules, firewallRulesOk := d.GetOk("firewall_rules"); firewallRulesOk {
+		fwrMap = generateForwardingRules(firewallRules)
```
Comment (mamclaughlin) [inline vultr/resource_vultr_load_balancer.go]: change function name to generateFirewallRules

File: website/docs/d/load_balancer.html.markdown [hunk 3]
```diff
@@ -51,6 +51,7 @@ The following attributes are exported:
 * `ipv6` - IPv6 address for your load balancer.
 * `health_check` - Defines the way load balancers should check for health. The configuration of a `health_check` is listed below.
 * `forwarding_rules` - Defines the forwarding rules for a load balancer. The configuration of a `forwarding_rules` is listened below.
+* `source` - (Required) IP address with subnet that is allowed through the firewall. You may also pass in `cloudflare` which will allow only CloudFlares IP range.
```
Comment (mamclaughlin) [inline website/docs/d/load_balancer.html.markdown]: this is defined in the firewall_rules below

File: website/docs/d/load_balancer.html.markdown [hunk 4]
```diff
@@ -67,4 +68,9 @@ The following attributes are exported:
...
 * `frontend_port` - Port on load balancer side.
 * `backend_protocol` - Protocol on instance side. Possible values: "http", "https", "tcp".
-* `target_port` - Port on instance side.
\ No newline at end of file
+* `target_port` - Port on instance side.
+
+`firewall_rules` supports the following
+* `frontend_port` - (Required) Port on load balancer side.
+* `ip_type` - (Required) The type of ip this rule is - may be either v4 or v6.
+* `source` - (Required) IP address with subnet that is allow through the firewall. You may also pass in `cloudflare` which will allow only CloudFlares IP range.
```
Comment (mamclaughlin) [inline website/docs/d/load_balancer.html.markdown]: allow through => allowed through

File: website/docs/d/load_balancer.html.markdown (hunk 3 above)
Comment (mamclaughlin) [reply website/docs/d/load_balancer.html.markdown]: add private_network

File: website/docs/r/private_network.html.markdown [hunk 5]
```diff
@@ -6,7 +6,7 @@ description: |-
   Provides a Vultr private network resource. This can be used to create, read, and delete private networks on your Vultr account.
 ---
 
-# vultr_network
+# vultr_private_network
```
Comment (mamclaughlin) [inline website/docs/r/private_network.html.markdown:9]: other references to vultr_network are still lingering, new pr for this? or just update the other references

Comment (ddymko) [reply website/docs/r/private_network.html.markdown:9]: I'll clean these up while I am here

**Review by mamclaughlin — APPROVED** [review]

**Review by mamclaughlin — APPROVED** [review]

**Pull request merged**